src/rust/aggregator/src/data/user_history.rs

```diff
@@ -59,6 +59,9 @@ impl Data for UserHistory {
             .begin()
             .await
             .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+        transaction.execute("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ;")
```

Comment:
Noting here that for postgres this is effectively equal to `SERIALIZABLE`

https://www.postgresql.org/docs/current/transaction-iso.html